config/file.go

```diff
@@ -30,7 +30,15 @@ type FileStore struct {
 }
 
 // NewFileStore creates a new instance of a config store backed by the given file path.
-func NewFileStore(path string) (fs *FileStore, err error) {
+func NewFileStore(path string, createFileIfNotExists bool) (fs *FileStore, err error) {
+	if createFileIfNotExists {
+		file, err2 := os.OpenFile(path, os.O_RDONLY|os.O_CREATE, 0644)
```

Comment:
Should we not do this in the `resolveConfigFilePath` function itself? The path passed here is not necessarily the final resolved path.

I am wondering if the check should be placed after the `resolvedPath` is computed. And then we check:

```go
  f, err := os.Open(resolvedPath)
  if err != nil && errors.Is(err, fs.ErrNotExist) {
    if createFileIfNotExists {
     // create file
    } else {
     return err
    }
   f.Close()
  }

```

And if we are doing this after `resolveConfigFilePath`, there might not be a need to modify it.

Reply:
Makes sense, will update accordingly.